pkg/querier/queryrange/limits.go

```diff
@@ -102,10 +103,11 @@ type UserIDTransformer func(context.Context, string) string
 type cacheKeyLimits struct {
 	Limits
 	transformer UserIDTransformer
+	iqo         util.IngesterQueryOptions
 }
 
 func (l cacheKeyLimits) GenerateCacheKey(ctx context.Context, userID string, r resultscache.Request) string {
-	split := l.QuerySplitDuration(userID)
+	split := SplitIntervalForTimeRange(l.iqo, l.Limits, l.QuerySplitDuration, []string{userID}, time.Now().UTC(), r.GetEnd().UTC())
```

Comment:
need not be part of this pr, but something I noticed during the review.

https://github.com/grafana/loki/blob/a50b9699c3fb93a5b92f5de248ccf84430c5a36c/pkg/storage/chunk/cache/resultscache/cache.go#L118

we could pass the tenant list down to `GenerateCacheKey` which I believe atm does not consider multi-tenant queries